mu-plugins/central-hub/tests/phpunit/integration/meta-data/getCustomFieldsValues.php

```diff
@@ -0,0 +1,183 @@
+		add_post_meta( $this->post, 'venue-name', 'Carnegie Hall' );
+
+		// Check database that post meta was added.
+		$this->assertSame( '10-04-2019', get_post_meta( $this->post, 'event-date', true ) );
+		$this->assertSame( '19:30:00', get_post_meta( $this->post, 'event-time', true ) );
+		$this->assertSame( 'Carnegie Hall', get_post_meta( $this->post, 'venue-name', true ) );
+
+		$expected_custom_fields = [
+			'event-date' => '10-04-2019',
+			'event-time' => '19:30:00',
+			'venue-name' => 'Carnegie Hall',
+		];
+
+		$this->assertSame( $expected_custom_fields, get_custom_fields_values( $this->post, 'events', $config ) );
+
+		// Clean up database.
+		delete_post_meta( $this->post, 'event-date' );
+		delete_post_meta( $this->post, 'event-time' );
+		delete_post_meta( $this->post, 'venue-name' );
+	}
+
+	/**
+	 * Test get_custom_fields_values() should return custom fields default values when post meta is not in database.
+	 */
+	public function test_should_return_custom_fields_default_values_when_post_meta_is_not_in_database() {
+		$config = [
+			'custom_fields' => [
+				'event-date' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'event-time' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'venue-name' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+			],
+		];
+
+		// Check that database does not contain post meta.
+		$this->assertSame( '', get_post_meta( $this->post, 'event-date', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'event-time', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'venue-name', true ) );
+
+		$expected_custom_fields = [
+			'event-date' => '',
+			'event-time' => '',
+			'venue-name' => '',
+		];
+
+		$this->assertSame( $expected_custom_fields, get_custom_fields_values( $this->post, 'events', $config ) );
+	}
+
+	/**
+	 * Test get_custom_fields_values() should register a callback_and return true with callback priority level.
+	 */
+	public function test_filter_registers_callback_and_returns_true_with_cb_priority_level() {
+		// Register anonymous callback to filter $tag with priority of 20 and 3 arguments.
+		add_filter( 'filter_meta_box_custom_fields', function ( $custom_fields ) {
+			$expected_callback_args = [
+				0 => [
+					'event-date' => '',
+					'event-time' => '',
+					'venue-name' => '',
+				],
+				1 => 'events',
+				2 => $this->post,
+			];
+
+			$this->assertSame( $expected_callback_args, func_get_args() );
+
+			$update_custom_fields = [
+				0 => [
+					'event-date' => '10-12-2019',
+					'event-time' => '19:30:00',
+					'venue-name' => 'Peabody Opera House',
+				]
+			];
+
+			$this->assertSame( $update_custom_fields, func_get_args() );
+		}, 20, 3 );
+
+		// Check if any callback has been registered to the filter hook
+		$this->assertTrue( has_filter( 'filter_meta_box_custom_fields' ) );
+
+		// Clean up.
+		$this->assertTrue( remove_all_filters( 'filter_meta_box_custom_fields', 20 ) );
```

Comment:
@rgadon107 There's no need to assert true here. Why? `remove_all_filters` is _not_ an expected behavior within the `get_custom_fields_values()` function. Rather, you are using it to unregister (unhook) the callback from the filter that's within this function.

Does this make sense?

Reply:
@hellofromtonya

>Does this make sense?

Yes. I thought I needed to `assertTrue` to confirm that the filter hook actually was removed. I'll remove it since it's not needed.